### ISSUE TITLE
The error for invalid tslint.json files could be prettier

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1681]
As mentioned by @JKillian in #1676, it would be nice to have some sort of `validate` function and/or interface(s) generated for each rule. If someone uses an invalid `tslint.json` perhaps TSLint should complain? We can define invalid as having unknown or incorrect (e.g. a string instead of a number) options for a rule.

**Comment by JKillian** [comment]
Perhaps a duplicate of https://github.com/palantir/tslint/issues/889? (Should've mentioned it in the other thread, sorry!) But maybe this is different - validating the whole file vs. just validating the rule options?

**Comment by JoshuaKGoldberg** [comment]
Validating the whole file could be considered separate from the individual rules, sure. 

When you have technically invalid JSON the error isn't completely pretty:

```cmd
tslint --config tslint.json index.ts

undefined:4
    "rules": {
    ^

SyntaxError: Unexpected string in JSON at position 33
    at Object.parse (native)
    at loadConfigurationFromPath (C:\Code\tslint\lib\configuration.js:130:31)
    at Object.findConfiguration (C:\Code\tslint\lib\configuration.js:70:12)
    at LintCli.processFiles (C:\Code\tslint\lib\lintCli.js:43:49)
    at processFiles (C:\Code\tslint\lib\tslint-cli.js:132:30)
    at Object.<anonymous> (C:\Code\tslint\lib\tslint-cli.js:176:1)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
```

...but yes, this is otherwise mostly a dup of #889.
